src/utils/hash-utils.ts

```diff
@@ -28,7 +28,7 @@ export const makeHashFromObject = (params: { [paramKey: string]: string }) => {
     return ""
   }
   return entries
-    .map(([key, value]) => `${key}=${encodeURIComponent(value)}`)
+    .map(([key, value]) => value !== undefined ? `${key}=${encodeURIComponent(value)}` : key)
```

Comment:
personally, i find removing unnecessary negations to be more readable, like:
```suggestion
    .map(([key, value]) => value === undefined ? key : `${key}=${encodeURIComponent(value)}`)
```
Which removes `not not` condition for `key`. But if for you it's more readable, it's ok :)